vnext/Shared/HermesRuntimeHolder.cpp

```diff
@@ -3,31 +3,108 @@
 
 #include "pch.h"
 
-#include <hermes/hermes.h>
+#include <memory>
 #include <mutex>
+
+#include <JSI/decorator.h>
+#include <cxxreact/MessageQueueThread.h>
+#include <cxxreact/SystraceSection.h>
+#include <hermes/hermes.h>
 #include "HermesRuntimeHolder.h"
 
+#if defined(HERMES_ENABLE_DEBUGGER)
+#include <hermes/inspector/chrome/Registration.h>
+#endif
+
 using namespace facebook;
 
 namespace facebook {
 namespace react {
 
+namespace {
+
+std::unique_ptr<facebook::hermes::HermesRuntime> makeHermesRuntimeSystraced(
+    const ::hermes::vm::RuntimeConfig &runtimeConfig) {
+  SystraceSection s("HermesExecutorFactory::makeHermesRuntimeSystraced");
+  return hermes::makeHermesRuntime(runtimeConfig);
+}
+
+#ifdef HERMES_ENABLE_DEBUGGER
+class HermesExecutorRuntimeAdapter : public facebook::hermes::inspector::RuntimeAdapter {
```

Comment:
>HermesExecutorRuntimeAdapter [](http://example.com/codeflow?start=6&length=28)

"final": since it is a class with virtual members and we do not need to inherit from it, it is better to add 'final' to allow C++ compiler optimize code in some cases.